x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -553,10 +406,20 @@ private GetAliasesRequest createGetAliasesRequest(FieldCapabilitiesResponse resp
             }
 
             Map<String, FieldCapabilities> types = new LinkedHashMap<>(entry.getValue());
-            // apply verification and possibly remove the "duplicate" CONSTANT_KEYWORD field type
-            final InvalidMappedField invalidField = validityVerifier.apply(fieldName, types);
-            // apply verification for fields belonging to index aliases
-            Map<String, InvalidMappedField> invalidFieldsForAliases = getInvalidFieldsForAliases(fieldName, types, aliases);
+            Map<String, InvalidMappedField> invalidFields = fieldValidator.validateField(fieldName, types, aliases);
+
+            // update the list of invalid fields for each index alias to be used to clean up the hierarchy of fields at the end
+            if (aliases != null && aliases.isEmpty() == false ) {
```

Comment:
I don't follow - why is a cleanup needed? We should not ignore conflicting fields rather point out that they are invalid, that is they exist but cannot be used due to conflicts.
Removing them means any query that refers to them will fail with field not found instead of field invalid.

Also, the algo could be improved so that when a field, we change its children as invalid so any further checks on them are not needed.

Reply:
This branch of code - the one where InAliasesFieldValidator is used - is only used for `SYS COLUMNS` (aka separate mappings). There are no projections or aggregations queries making use of the fields generated in this case. The output goes only to `SYS COLUMNS` display.

The other branch of code - the one where InIndicesFieldValidator is used - is the one also used in queries. That's for a merged mapping and the validation errors are relevant there.

Regarding the algorithm, we won't know a certain child field is valid or not unless we build its entire hierarchy. The algorithm starts from children and goes up the hierarchy until all parents are built. If a second child belonging to the same (faulty) parent starts being built, we won't know we are building its hierarchy in vain until we hit that (faulty) parent. Determining if a child building is worth starting or not has a complexity overhead that might not be better than having the clean-up step at the end of the algorithm.

Reply:
> We should not ignore conflicting fields rather point out that they are invalid, that is they exist but cannot be used due to conflicts.

Not sure if relevant, but we currently don't do this, do we? `SYS COLUMNS TABLE LIKE 'test%'` (from #53445) will only report the non-conflicting fields. Don't know if that's intended, though.

Reply:
@bpintea that's true. Talked to @costin about this some more and the final solution will not report conflicting fields. For completeness sake I will create a meaningful error message and will log the conflicting fields if debugging is enabled (not to lose or not to hide this information in all cases).